### PR TITLE
ci: Add pre-commit configuration

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

**Comment by adamamer20** [comment]
Thanks. Shouldn't Black also be added?

**Comment by adamamer20** [comment]
Also, related to #15, do you think it would make sense to use tox/nox to ensure maximum compatibility by running tests with multiple python versions?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.4.3
+  hooks:
+    # Run the linter.
+    - id: ruff
+      types_or: [ python, pyi, jupyter ]
+      args: [ --fix ]
+    # Run the formatter.
+    - id: ruff-format
```
Comment (rht) [inline .pre-commit-config.yaml:14]: > Thanks. Shouldn't Black also be added?

`ruff format` is a fast Black replacement.

**Comment by rht** [comment]
> Also, related to https://github.com/adamamer20/mesa-frames/pull/15, do you think it would make sense to use tox/nox to ensure maximum compatibility by running tests with multiple python versions?

The GitHub Actions CI testing (which I will set up after #15 is merged) supports running multiple Python versions declaratively. You can test only 1 version on your local machine, for faster feedback loop, and leave testing multiple versions to the CI.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (adamamer20) [reply .pre-commit-config.yaml:14]: ah you're right, thank you.

**Review by adamamer20 — APPROVED** [review]

**Pull request merged**